### PR TITLE
Update to RockyLinux 9 and Ubuntu 22.04

### PR DESCRIPTION
**Pull request by khaledk2**
It has passed on https://github.com/khaledk2/ansible-role-postgresql-client/actions/runs/6226134735

I have got this we get this error: 
https://github.com/khaledk2/ansible-role-postgresql-client/actions/runs/6225843952/job/16897118456#step:5:242

I have tested the command on a RockyLinux 9 docker container and worked fine without getting this error. I have tested also on a machine running with RockyLinux 9 and worked without any errors.

So, I have added `disable_gpg_check: yes` to `postgres | setup dnf repository` .

### CODE REVIEW

**Review by sbesson — COMMENTED** [review]
Thanks @khaledk2, most of the changes are very straighforward. The only debatable one as noted in the PR description is the disabling of the `gpg_check`.

Looking at https://download.postgresql.org/pub/repos/yum/, there seem to be a set of GPG keys for each version of PSQL. Thoughts about trying to replicate the logic of the Debian task and make use https://docs.ansible.com/ansible/latest/collections/ansible/builtin/rpm_key_module.html to import the key before to installing the PSQL repositories?

**Review by sbesson — CHANGES_REQUESTED** [review]
Thanks @khaledk2 the last commits look very promising.

Can we combine the series of tasks introduced in https://github.com/ome/ansible-role-postgresql-client/pull/8/commits/850a85c3f75993bcde6729e7deec0641856340e6 into a single one using the `{{ postgresql_version }}` variable? I believe all these keys are identical anyways and we might be able to use https://download.postgresql.org/pub/repos/yum/RPM-GPG-KEY-PGDG

**Comment by khaledk2** [comment]
Thank you @sbesson,  yes, I have noticed that  all the keys are identical, so only one key ( https://download.postgresql.org/pub/repos/yum/RPM-GPG-KEY-PGDG) is imported now

File: tasks/redhat.yml [hunk 1]
```diff
@@ -1,15 +1,21 @@
 ---
+- name: Import a key for poastgres
```
Comment (sbesson) [inline tasks/redhat.yml]:
```suggestion
- name: Import a key for postgres
```

**Review by sbesson — COMMENTED** [review]
One typo. Otherwise looks good to me (and I assume `ansible-role-postgresql` can use exactly the same changes)

**Review by sbesson — APPROVED** [review]

**Comment by jburel** [comment]
We could consider adding support for  pg 15.

**Pull request merged**